data-workflows/activity/install_activity_model.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import time
+from datetime import datetime, timezone
+from enum import Enum
+from typing import Callable, List, Union
+import os
+
+from pynamodb.models import Model
+from pynamodb.attributes import UnicodeAttribute, NumberAttribute
+
+from utils.utils import get_current_timestamp
+
+LOGGER = logging.getLogger()
+
+
+def to_utc_timestamp_in_millis(timestamp: datetime) -> int:
+    return int(timestamp.replace(tzinfo=timezone.utc).timestamp() * 1000)
+
+
+class InstallActivityType(Enum):
+
+    def __new__(cls, value, timestamp_formatter, type_timestamp_formatter):
+        install_activity_type = object.__new__(cls)
+        install_activity_type._value = value
+        install_activity_type.timestamp_formatter = timestamp_formatter
+        install_activity_type.type_timestamp_formatter = type_timestamp_formatter
+        return install_activity_type
+
+    DAY = (1, to_utc_timestamp_in_millis, lambda timestamp: f'DAY:{timestamp.strftime("%Y%m%d")}')
+    MONTH = (2, to_utc_timestamp_in_millis, lambda timestamp: f'MONTH:{timestamp.strftime("%Y%m")}')
+    TOTAL = (3, lambda timestamp: None, lambda timestamp: 'TOTAL:')
+
+    def get_timestamp_formatter(self) -> Callable[[datetime], Union[int, None]]:
+        return self.timestamp_formatter
+
+    def get_query_timestamp_projection(self) -> str:
+        return '1' if self is InstallActivityType.TOTAL else f"DATE_TRUNC('{self.name}', timestamp)"
+    
+    def get_type_timestamp_formatter(self) -> Callable[[datetime], str]:
+        return self.type_timestamp_formatter
+
+    def get_timestamp_query_formatter(self) -> Callable[[datetime], datetime]:
+        return (lambda ts: ts) if self is InstallActivityType.DAY else (lambda ts: ts.replace(day=1))
```

Comment:
capturing explicitly here, you could use `functools.partial`:
```suggestion
        return (lambda ts: ts) if self is InstallActivityType.DAY else partial(datetime.replace, day=1)
```

Reply:
This is awesome. I didn't know we could do this in python as well.